File: 53. Maximum Subarray/maxSubArray.java
class maxSubArray {
    public int solution(int[] nums) {
        int sum = 0, res = nums[0];
        for (int n: nums) {
            if (sum > 0) {
                sum = sum + n;
            }
            else {
                sum = n;
            }
            res = Math.max(sum, res);
        }
        return res;
    }
}

File: 53. Maximum Subarray/maxSubArray.py
class maxSubArray:
    def solution(self, nums: List[int]) -> int:
        sum = 0
        res = nums[0]
        for n in nums:
            if sum > 0:
                sum = sum + n
            else:
                sum = n
            res = max(sum, res)
        return res